tests/utils/test_module_inputs_wrapper.py

```
@@ -0,0 +1,93 @@
+    ek_reader_result = WRAPPED_XR_ONES_LIKE(EK_GRIB_READER)
+    assert isinstance(ek_reader_result, XR_TYPES)
+    assert ek_reader_result.equals(xr.ones_like(EK_GRIB_READER.to_xarray()))
+
+
+def test_transform_function_inputs_wrapper_to_xarray():
+    # EK XarrayWrapper object
+    ek_wrapper_result = WRAPPED_XR_ONES_LIKE(EK_XARRAY_WRAPPER)
+    assert isinstance(ek_wrapper_result, XR_TYPES)
+    assert ek_wrapper_result.equals(xr.ones_like(EK_XARRAY_WRAPPER.data))
+    # EK NumpyWrapper object
+    ek_wrapper_result = WRAPPED_XR_ONES_LIKE(EK_NUMPY_WRAPPER)
+    assert isinstance(ek_wrapper_result, XR_TYPES)
+    assert ek_wrapper_result.equals(xr.ones_like(TEST_DA))
+
+
+def test_transform_function_inputs_reader_to_numpy():
+    # Test with Earthkit.data GribReader object
+    assert WRAPPED_NP_MEAN(EK_GRIB_READER) == np.mean(EK_GRIB_READER.to_numpy())
+    assert isinstance(WRAPPED_NP_MEAN(EK_GRIB_READER), np.float64)
+
+
+def test_transform_function_inputs_wrapper_to_numpy():
+    # Test with Earthkit.data XarrayWrapper object
+    ek_object_result = WRAPPED_NP_MEAN(EK_XARRAY_WRAPPER)
+    assert ek_object_result == np.mean(TEST_DS)
+    assert isinstance(ek_object_result, type(EK_XARRAY_WRAPPER.data))
+
+    # Test with Earthkit.data NumpyWrapper object
+    ek_object_result = WRAPPED_NP_MEAN(EK_NUMPY_WRAPPER)
+    assert ek_object_result == np.mean(TEST_NP)
+    assert isinstance(ek_object_result, np.float64)
+
+
+def test_transform_function_inputs_xarray_to_numpy():
+    # Test with xarray.DataArray object
+    assert WRAPPED_NP_MEAN(TEST_DA) == np.mean(TEST_NP)
+    assert WRAPPED_NP_MEAN(TEST_DA) == np.mean(TEST_DA)
+    assert isinstance(WRAPPED_NP_MEAN(TEST_DA), xr.DataArray)
+
+
+def test_transform_function_inputs_pandas_to_numpy():
+    # Test with pandas.DataFrame object
+    assert WRAPPED_NP_MEAN(TEST_DF) == np.mean(TEST_NP)
+    assert WRAPPED_NP_MEAN(TEST_DF) == np.mean(TEST_DF)
+    assert isinstance(WRAPPED_NP_MEAN(TEST_DF), np.float64)
```

I suggest you also add a test for the case when  `kwarg_types` or/and `convert_types` are invalid or do not match.

---

I'm not sure I understand the test you want.

If the input is not transformable to an expected type it is parsed unchanged the if that is invalid then the wrapped function/module would raise the error. Do you want me to include a test that asserts that an Error from the wrapped function is raised?

---

But to add to this, whilst adding the transform_module_inputs tests I ended up adding a lot more edge cases, which cover when we use type-setting of the wrapped methods and do not use kwarg_types/convert_types

---

> I'm not sure I understand the test you want.
> 
> If the input is not transformable to an expected type it is parsed unchanged, then if that is invalid then the wrapped function/module would raise the error. Do you want me to include a test that asserts that an Error from the wrapped function is raised?

Yes, what I suggest is to add some tests to see if the right exception is raised when the args are incorrect. .E.g. the following call should raise an exception since b is not an arg of np.mean():

```python
 module_inputs_wrapper.transform_function_inputs(
    np.mean,
    kwarg_types={"b": np.ndarray}
)
```

---

This doesn't raise an exception, and I don't think it should (acknowledged this is a subjective opionion). Imposing a restraint here would mean that it would make it tricker to transform a series of functions with a single configuration (like I do with the transform_module_inputs)